src/main/java/io/stargate/sgv2/jsonapi/service/embedding/gateway/EmbeddingGatewayClient.java

```diff
@@ -0,0 +1,162 @@
+    this.baseUrl = baseUrl;
+    this.embeddingService = embeddingService;
+    this.vectorizeServiceParameter = vectorizeServiceParameter;
+  }
+
+  /**
+   * Vectorize the given list of texts
+   *
+   * @param texts List of texts to be vectorized
+   * @param apiKeyOverride API key sent as header
+   * @param embeddingRequestType Type of request (INDEX or SEARCH)
+   * @return
+   */
+  @Override
+  public Uni<List<float[]>> vectorize(
+      List<String> texts,
+      Optional<String> apiKeyOverride,
+      EmbeddingRequestType embeddingRequestType) {
+    Map<String, EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue>
+        grpcVectorizeServiceParameter = new HashMap<>();
+    if (vectorizeServiceParameter != null) {
+      vectorizeServiceParameter.forEach(
+          (key, value) -> {
+            if (value instanceof String)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setStrValue((String) value)
+                      .build());
+            else if (value instanceof Integer)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setIntValue((Integer) value)
+                      .build());
+            else if (value instanceof Float)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setFloatValue((Float) value)
+                      .build());
+            else if (value instanceof Boolean)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setBoolValue((Boolean) value)
+                      .build());
+          });
+    }
+    EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest embeddingRequest =
+        EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.newBuilder()
+            .setModelName(modelName)
+            .setDimensions(dimension)
+            .putAllParameters(grpcVectorizeServiceParameter)
+            .setInputType(
+                embeddingRequestType == EmbeddingRequestType.INDEX
+                    ? EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.InputType.INDEX
+                    : EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.InputType.SEARCH)
+            .addAllInputs(texts)
+            .build();
+
+    final EmbeddingGateway.ProviderEmbedRequest.ProviderContext.Builder builder =
+        EmbeddingGateway.ProviderEmbedRequest.ProviderContext.newBuilder()
+            .setProviderName(provider)
+            .setTenantId(tenant.orElse("default"));
```

Comment:
Maybe define constant for `DEFAULT_TENANT_ID`?

Not sure where this should be documented, but would be also good to explain briefly meaning -- does this mean defaulting or that tenant-id is not available and not used (which I think is the case)?
(could also use different token here -- empty String, or `"missing"` or something -- unless there is some actual default tenant-id that is used)